### PR TITLE
feat: animated review expando

### PR DESCRIPTION
**Pull request by cartcrom**
## Description

Adds expando that hides some swap line items in the review modal

Before:

https://github.com/Uniswap/interface/assets/39385577/a4b9bdfa-55f9-4fac-800e-d74586d97207

After:

https://github.com/Uniswap/interface/assets/39385577/f48bf870-a900-4918-8876-8e065e24f011

### CODE REVIEW

File: src/components/swap/SwapModalFooter.tsx [hunk 1]
```diff
@@ -36,6 +41,31 @@ const ConfirmButton = styled(ButtonError)`
...
+  margin-right: -6px;
+
+  padding: 16px;
+  min-width: fit-content;
+  white-space: nowrap;
+`
+
+function DropdownController({ open, onClick }: { open: boolean; onClick: () => void }) {
+  return (
+    <Row onClick={onClick} style={{ cursor: 'pointer' }}>
```
Comment (tinaszheng) [inline src/components/swap/SwapModalFooter.tsx:56]: if this is clickable it needs to be a `button` for a11y 👀

File: src/components/swap/SwapModalFooter.tsx [hunk 2]
```diff
@@ -66,17 +96,16 @@ export default function SwapModalFooter({
...
   return (
     <>
+      <DropdownController open={showMore} onClick={() => setShowMore(!showMore)} />
       <DetailsContainer gap="md">
         <SwapLineItem {...lineItemProps} type={SwapLineItemType.EXCHANGE_RATE} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.PRICE_IMPACT} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MAX_SLIPPAGE} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MAXIMUM_INPUT} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MINIMUM_OUTPUT} />
+        <AdvancedSwapDetails {...{ trade, allowedSlippage, open: showMore }} />
```
Comment (tinaszheng) [inline src/components/swap/SwapModalFooter.tsx:108]: i think you should make these into individual props for readability

**Review by tinaszheng — CHANGES_REQUESTED** [review]

**Review by gnewfield — APPROVED** [review]
LGTM

File: src/components/swap/SwapModalFooter.tsx (hunk 2 above)
Comment (cartcrom) [reply src/components/swap/SwapModalFooter.tsx:108]: OOF i meant to refactor this ty

Comment (cartcrom) [reply src/components/swap/SwapModalFooter.tsx:108]: embarassed

File: src/components/swap/SwapModalFooter.tsx [hunk 3]
```diff
@@ -141,3 +170,43 @@ export default function SwapModalFooter({
...
+  const animatedProps = useSpring({
+    animatedOpacity: open ? 1 : 0,
+    config: { duration: ms('300ms'), easing: easings.easeOutSine },
+    delay,
+  })
+
+  return <SwapLineItem {...props} {...animatedProps} />
+}
+
+function AdvancedSwapDetails(props: { trade: InterfaceTrade; allowedSlippage: Percent; open: boolean }) {
```
Comment (just-toby) [inline src/components/swap/SwapModalFooter.tsx:186]: we have another `AdvancedSwapDetails` component defined in the project already - might get confusing to have both? can we rename this something like `ExpandableLineItems` ?

**Review by just-toby — APPROVED** [review]
one suggestion

File: src/components/swap/SwapModalFooter.tsx (hunk 3 above)
Comment (cartcrom) [reply src/components/swap/SwapModalFooter.tsx:186]: excellent point as always edward

File: src/components/swap/SwapModalFooter.tsx (hunk 2 above)
Comment (tinaszheng) [reply src/components/swap/SwapModalFooter.tsx]: ay lmao

**Review by tinaszheng — APPROVED** [review]

**Pull request merged**